healthz/README.md

```diff
@@ -185,6 +193,14 @@ or to examine the state of a protocol.  These commands will likely be
 considered "service impacting" and should have a clear security ACL restricting
 their use during normal operations of the device.
 
+The `CheckRequest` message includes an `event_id` field. When populated this
+indicates that the `Check` should be performed for an event that has already
+occurred within the system. The device should trigger artifact collection of
+those artifacts that were not automatically collected. A `CheckRequest` for
+a previous `event_id` MUST NOT overwrite previous artifacts that were collected
+at the time of the event. The artifacts that are collected MUST be returned
+in the artifact list for the event when reported by the `Get` or `List` RPCs.
```

Comment:
@robshakir I wonder how this is supposed to work, given that (I think) the health of a component can't be checked for the past events as the state of the component may already be HEALTHY or the artifacts collected won't contain the same state that were relevant at the time of a  healthy->unhealthy  transition.

I am inclined to not stall the PR though, as currently in our system we don't anticipate having expensive artifact collection processes which would require `Check` to work on a past `event_id`...

Reply:
synced up offline
copying for historical purposes

> you call Check and it gathers some artifact that is based on the current state. This might include historical logs in the case of dumping some database, or retrieving some coredump from the secondary control plane.
But there's no guarantee that it captured the state at the time. For that, we need the artifacts that are automatically created to be complete.